File: C++/bubblesort.cpp
#include <bits/stdc++.h>
using namespace std;

void swap(int *a, int *b) {
    int temp;
    temp = *a;
    *a = *b;
    *b = temp;
}

void BubbleSort(int a[], int n) {
    for(int j=0;j<n;j++) {
        for(int i=0;i<n-1;i++) {
            if(a[i] > a[i+1]) {
                swap(&a[i], &a[i+1]);
            }
        }
    }

    for(int k=0;k<n;k++) { cout << a[k] << " "; }
}

int main() {

    int arr[] = {5,4,1,2,3};
    int n = 5;
    BubbleSort(arr, n);
    return 0;
}

File: javaScript/Algorithms/bubblesort.js
function bubbleSort(arr) {
    var isSwapping
    for (let i = arr.length; i > 0; i--) {
        isSwapping = true
        for (let j = 0; j < i-1; j++) {
            if (arr[j] > arr[j + 1]) {
                let temp = arr[j]
                arr[j] = arr[j + 1]
                arr[j + 1] = temp
                isSwapping = false
            }
        }
        if(isSwapping) break    // For optimization(exit from loop if there is nothing to swap)
    }
    return arr
}

console.log(bubbleSort([55,23,76,1,20,36]))
